### PR TITLE
Add util functions to clustifyr

### PR DESCRIPTION
**Pull request by SidhantPuntambekar**
Add util functions such as check_raw_counts, build_atlas, and append_genes from scRNA-seq-Ref-Matrix repo to clustifyr. Should solve #358.

### CODE REVIEW

**Comment by raysinensis** [comment]
One thing to note: Bioconductor really does not like lines with 80+ characters, so it might be good to move comment lines to their own line

File: R/utils.R [hunk 1]
```diff
@@ -1862,3 +1862,147 @@ find_rank_bias <- function(mat,
...
+#' @param counts_matrix Count matrix containing scRNA-seq read data
+#' @param max_log_value Static value to determine if a matrix is normalized
+#' @return String either raw counts, log-normalized or normalized
+#' @example
+#' mat <- append_genes(
+#' gene_vector = human_genes_10x,
+#' ref_matrix = cbmc_ref 
+#' ) 
+#' @export
+check_raw_counts <- function(counts_matrix, max_log_value = 50)
```
Comment (raysinensis) [inline R/utils.R:1919]: apparently this does not recognize sparse matrix...
compare
```
check_raw_counts(counts_matrix = as.matrix(pbmc_matrix_small), max_log_value = 50)
check_raw_counts(counts_matrix = pbmc_matrix_small, max_log_value = 50)
```

test with something like this: 
```
if (is(pbmc_matrix_small, 'sparseMatrix')) {
  pbmc_matrix_small <- as.matrix(pbmc_matrix_small)
}
```

File: R/utils.R [hunk 2]
```diff
@@ -1862,3 +1862,147 @@ find_rank_bias <- function(mat,
...
+        names(ref_mats) <- basename(matrix_fns) %>% stringr::str_remove(".rds$")
+    } else {
+        names(ref_mats) <- names(matrix_fns)
+    }
+    
+    # iterate over list and get new matrices
+    new_mats <- list()
+    for(i in seq_along(ref_mats)){
+        # standardize genes in matrix
+        mat <- clustifyr:::append_genes(gene_vector = genesVector,
```
Comment (raysinensis) [inline R/utils.R:1983]: no longer need "clustifyr:::", sorry for the confusion

File: R/utils.R (hunk 1 above)
Comment (SidhantPuntambekar) [reply R/utils.R:1919]: See bd9647e

File: R/utils.R (hunk 2 above)
Comment (SidhantPuntambekar) [reply R/utils.R]: See f20602a

File: R/utils.R (hunk 1 above)
Comment (SidhantPuntambekar) [reply R/utils.R:1919]: Also see 4992003

File: R/utils.R [hunk 3]
```diff
@@ -1862,3 +1862,150 @@ find_rank_bias <- function(mat,
...
+#' @param geneVector char vector with gene names
+#' @param ref_matrix Reference matrix containing cell types vs. 
+#' gene expression values
+#' @return Reference matrix with union of all genes
+append_genes <- function(gene_vector, ref_matrix)
+{
+    rownamesGSEMatrix <- rownames(ref_matrix) 
+    #Get rownames from GSEMatrix (new GSE file)
+    
+    rowCountHumanGenes <- nrow(gene_vector)
```
Comment (raysinensis) [inline R/utils.R:1878]: this should be length(gene_vector)

Comment (SidhantPuntambekar) [reply R/utils.R]: See 55f469c

**Pull request merged**